app/Http/Controllers/OAuthController.php

```diff
@@ -2,20 +2,67 @@
 
 namespace App\Http\Controllers;
 
-use App\Models\User;
-use CDash\Middleware\OAuth2\OAuth2Interface;
 use Illuminate\Http\RedirectResponse;
 use Illuminate\Http\Request;
 use Illuminate\Support\Collection;
 use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\Hash;
+use Illuminate\Support\Facades\Log;
+use Illuminate\Support\Str;
+use Laravel\Socialite\Facades\Socialite;
+use Illuminate\Routing\Redirector;
+use Symfony\Component\HttpFoundation\RedirectResponse as symfonyResponse;
 use RuntimeException;
 
+use App\Models\User;
+use App\Http\Controllers\Auth\RegisterController;
+use CDash\Middleware\OAuth2\OAuth2Interface;
+
 /**
  * Class OAuthController
  * @package App\Http\Controllers
  */
 final class OAuthController extends AbstractController
 {
+    public function socialite(Request $request, string $service): symfonyResponse
+    {
+        return Socialite::driver($service)->redirect();
+    }
+
+    public function callback(string $service, User $user_table): RedirectResponse|Redirector
```

Comment:
See previous comment about the `User` model.  If you didn't see the comments, it might be because GitHub collapsed them by default.  I don't believe you need to pass the User model here..